### PR TITLE
fix(binder): bind struct field access with bind column

### PR DESCRIPTION
**Pull request by st1page**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Unify the bind column path for bind field_column of composite type. Now when binding the expr in the `FieldIdentifier`(`(table.column).field`), binder will call `bind_column` so that other features can works well such as extracting schema name and binding correlated columns. 
Also,
- open the create schema statment in planner test
- add some test for composite type field accessing.
## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
fix https://github.com/risingwavelabs/risingwave/issues/6772

### CODE REVIEW

**Review by xxchan — COMMENTED** [review]
The tests look good. Impl not checked yet

File: e2e_test/batch/duckdb/select/test_multi_column_reference.slt.part [hunk 1]
```diff
@@ -118,8 +117,38 @@ statement ok
...
+----
+42
+
+statement ok
+DROP TABLE t.t;
+
+statement ok
+DROP SCHEMA t;
+
+# (original test's syntax is different): https://github.com/risingwavelabs/risingwave/issues/6772
```
[the remark this answers is not in the document]
Comment (st1page) [reply e2e_test/batch/duckdb/select/test_multi_column_reference.slt.part]: fixed

**Comment by xxchan** [comment]
Suggested edit:
```diff
e2e_test/batch/duckdb/select/test_multi_column_reference.slt.part --- Text
141 141 ----142 142 42143 143 144 ... statement ok145 ... DROP TABLE t.t;146 144 147 145 statement ok148 146 DROP SCHEMA t;

```

File: src/frontend/src/binder/struct_field.rs [hunk 2]
```diff
@@ -13,72 +13,48 @@
...
-                .get_column_binding_index(&table_name, &column_name)?;
-            let binding = &self.context.columns[index];
-            let expr = InputRef::new(binding.index, binding.field.data_type.clone()).into();
-            Ok((expr, ids))
-        } else {
-            match expr {
-                Expr::Cast { expr, data_type } => {
-                    let cast = self.bind_cast(*expr, data_type)?;
-                    Ok((cast, ids))
+            _ => unreachable!(),
```
Comment (xxchan) [inline src/frontend/src/binder/struct_field.rs:35]: It's reachable: https://github.com/risingwavelabs/risingwave/issues/7011 😄 

Do we fix it here or later?

**Review by xxchan — APPROVED** [review]
LGTM (It's trivial 🤪). Added some comments & style refactor while understanding struct stuff.

File: src/frontend/src/binder/struct_field.rs (hunk 2 above)
Comment (xxchan) [reply src/frontend/src/binder/struct_field.rs]: It's a different issue. Let's take some time to consider it and merge this PR first.

**Pull request merged**